src/ioctls/device.rs

```diff
@@ -118,28 +176,36 @@ mod tests {
             fd: 0,
             flags: KVM_CREATE_DEVICE_TEST,
         };
-        // This fails on aarch64 as it does not use MPIC (MultiProcessor Interrupt Controller), it uses
-        // the VGIC.
+        // This fails on aarch64 as it does not use MPIC (MultiProcessor Interrupt Controller),
+        // it uses the VGIC.
         assert!(vm.create_device(&mut gic_device).is_err());
 
-        let device_fd = create_gic_device(&vm, KVM_CREATE_DEVICE_TEST);
+        let device_fd = create_gic_device(&vm, 0);
 
         // Following lines to re-construct device_fd are used to test
         // DeviceFd::from_raw_fd() and DeviceFd::as_raw_fd().
         let raw_fd = unsafe { libc::dup(device_fd.as_raw_fd()) };
         assert!(raw_fd >= 0);
         let device_fd = unsafe { DeviceFd::from_raw_fd(raw_fd) };
 
+        // Set some attribute that does not apply to VGIC, expect the test to fail.
         let dist_attr = kvm_bindings::kvm_device_attr {
             group: KVM_DEV_VFIO_GROUP,
             attr: u64::from(KVM_DEV_VFIO_GROUP_ADD),
             addr: 0x0,
             flags: 0,
         };
+        assert!(device_fd.has_device_attr(&dist_attr).is_err());
 
-        // We are just creating a test device. Creating a real device would make the CI dependent
-        // on host configuration (like having /dev/vfio). We expect this to fail.
-        assert!(device_fd.set_device_attr(&dist_attr).is_err());
-        assert_eq!(errno::Error::last().errno(), 25);
+        // Following attribute works with VGIC, they should be accepted.
+        let dist_attr = kvm_bindings::kvm_device_attr {
+            group: KVM_DEV_ARM_VGIC_GRP_CTRL,
+            attr: u64::from(KVM_DEV_ARM_VGIC_CTRL_INIT),
+            addr: 0x0,
+            flags: 0,
+        };
+
+        assert!(device_fd.has_device_attr(&dist_attr).is_ok());
+        assert!(device_fd.set_device_attr(&dist_attr).is_ok());
```

Comment:
Why change: The old code here was forked from X86 test case where a VFIO device was created with TEST flag, because creating a real device would make the test code depend on host configuration.
But in arm test case, what's created is a VGIC. We can create a "real" device to test with real result.